frame/transaction-payment/src/lib.rs

```diff
@@ -564,27 +583,57 @@ where
 		.map(|i| (fee, i))
 	}
 
-	/// Get an appropriate priority for a transaction with the given length and info.
+	/// Get an appropriate priority for a transaction with the given `DispatchInfo` and fee to be
+	/// paid (with `tip`).
 	///
-	/// This will try and optimise the `fee/weight` `fee/length`, whichever is consuming more of the
-	/// maximum corresponding limit.
+	/// Values returned by the function depend on the following:
+	/// 1. The average per-weight-unit tip to be payed.
+	/// 2. The dispatch class of the transaction.
 	///
-	/// For example, if a transaction consumed 1/4th of the block length and half of the weight, its
-	/// final priority is `fee * min(2, 4) = fee * 2`. If it consumed `1/4th` of the block length
-	/// and the entire block weight `(1/1)`, its priority is `fee * min(1, 4) = fee * 1`. This means
-	///  that the transaction which consumes more resources (either length or weight) with the same
-	/// `fee` ends up having lower priority.
+	/// For Normal (and Mandatory) dispatchables we simply take the amount of `tip` payed
+	/// for every weight unit, scaled up by `T::WeightTipRatio`.
+	/// Operational dispatchables get an additional `priority` increase, roughly matching
+	/// a tip with amount equal to the cost required to include transaction that consumes
+	/// half of the block weight.
 	fn get_priority(
-		len: usize,
 		info: &DispatchInfoOf<T::Call>,
-		final_fee: BalanceOf<T>,
+		tip: BalanceOf<T>,
 	) -> TransactionPriority {
-		let weight_saturation = T::BlockWeights::get().max_block / info.weight.max(1);
-		let max_block_length = *T::BlockLength::get().max.get(DispatchClass::Normal);
-		let len_saturation = max_block_length as u64 / (len as u64).max(1);
-		let coefficient: BalanceOf<T> =
-			weight_saturation.min(len_saturation).saturated_into::<BalanceOf<T>>();
-		final_fee.saturating_mul(coefficient).saturated_into::<TransactionPriority>()
+		let max_block = T::BlockWeights::get().max_block;
+		let bounded_weight = info.weight.max(1).min(max_block);
+		let weight_tip_ratio = T::WeightTipRatio::get();
+		let per_weight = |val| FixedU128::saturating_from_rational(val, bounded_weight)
+			.saturating_mul_int(weight_tip_ratio);
+
+		let tip_per_weight = per_weight(tip);
+
+		match info.class {
+			DispatchClass::Normal => {
+				// For normal class we simply take the `tip_per_weight`.
+				tip_per_weight
+			},
+			DispatchClass::Mandatory => {
+				// Mandatory extrinsics should be prohibited (e.g. by the [`CheckWeight`] extensions),
+				// but just to be safe let's return the same priority as `Normal` here.
+				tip_per_weight
+			},
+			DispatchClass::Operational => {
+				// A reasonable tip value to frontrun an `Operational` extrinsic.
+				// This value should be kept high enough to allow `Operational` extrinsics
+				// from getting in even during congested period, but at the same time low
```

Comment:
this is confusing to me, do you mean:
```suggestion
				// This value should be kept high enough to allow `Operational` extrinsics
				// to get in even during congested period, but at the same time low
```